web/src/App.tsx

```diff
@@ -0,0 +1,94 @@
+import { Box, Center, Flex } from "@chakra-ui/react";
+import React, { useState } from "react";
+import {
+  Navigate,
+  Route,
+  Routes,
+  useLocation,
+  useNavigate,
+} from "react-router-dom";
+import { useRecoilValue } from "recoil";
+
+import { useAnalytics } from "@/analytics";
+import { Footer } from "@/components/Footer";
+import { Loading } from "@/components/loader/Loader";
+import { Nav } from "@/components/navBar/Nav";
+import { useUpdateCityList } from "@/data/models/useUpdateCityList";
+import { connectionConfig } from "@/data/recoil";
+import { AnalyticsDashboard } from "@/pages/Analytics";
+import { Overview } from "@/pages/Configure";
+import { NotificationsMap } from "@/pages/Dashboard";
+import { HomePage } from "@/pages/HomePage";
+import { useConnectionState } from "@/view/hooks/hooks";
+
+function App() {
+  const loadingFallback = (
+    <Center height="100vh">
+      <Loading size="large" centered={true} />
+    </Center>
+  );
+
+  return (
+    <React.Suspense fallback={loadingFallback}>
+      <RoutesContainer />
+    </React.Suspense>
+  );
+}
+
+const RoutesContainer = () => {
+  const config = useRecoilValue(connectionConfig);
+  const { connected } = useConnectionState();
+  const { updateCityList } = useUpdateCityList(config);
+  const [redirectToHomePage, setRedirectToHomaePage] = useState(true); // This allow user to modify connection settings in configuration page without redirecting to Home page
```

Comment:
Will make the changes

Just a small question

Let's say I open

/configure

Now since I am not signed so it will take me to /
after adding credentials should it redirect to /dashboard or /configure ??

Reply:
Also there is one issue in using connected to identify the state

If you go to configure after you are connected and make even a small change in connection it simply redirects to homepage instead of staying there as we are dependent on connected and not on firstTimeConnected

I need another variable and react effected to get rid of this effect considering your design as well 

Do you think i should do that or are we good with it redirecting immediately

I personally feel it as bad exp

Reply:
> Now since I am not signed so it will take me to /
> after adding credentials should it redirect to /dashboard or /configure ??

it would be better if it takes you to /configure but it's ok to take you to /dashboard

> Do you think i should do that or are we good with it redirecting immediately

we shouldn't redirect

Reply:
I have made some changes to address your concern yet have the same behaviour let me know if that is good for this specific section